### ISSUE TITLE
Unable to configure decoder options

### ISSUE THREAD

**Issue by myisaak** [jimp-dev/jimp#939]
https://github.com/oliver-moran/jimp/blob/909f06eac36ef5ba36ebb1e0853fd006c4da3560/packages/core/src/utils/image-bitmap.js#L165

This library doesn't allow for configuring some decoders that take extra parameters. The line above shows this explicitly where only one argument is given instead of at least another options object. Decoders such as the `jpeg-js` one offer extra configurations in the form of another configuration that can be useful for a given context (see here google/eleventy-high-performance-blog#10).

My proposal fix would be to change the constructor function to allow an `decoderOptions` key that map mimetypes to the respective decoder options. For example:
```js
const decoderOptions = {
  "image/jpeg": {
    maxMemoryUsageInMB: 1024,
  }
};
Jimp.decode({ path: "filepath.jpg", decoderOptions, });
Jimp.decode("filepath.jpg", { decoderOptions }); // alternative
```

**Comment by hisham** [comment]
+1 any update or workaround for this?

**Comment by hisham** [comment]
You can do this currently by overriding the jpeg-js decoder jimp uses as follows:

`Jimp.decoders['image/jpeg'] = (data: Buffer) => JPEG.decode(data, { maxMemoryUsageInMB: 1024 });`

**Comment by hipstersmoothie** [comment]
:rocket: Issue was released in [`v1.6.0`](https://github.com/jimp-dev/jimp/releases/tag/v1.6.0) :rocket:

**Comment by sayo96** [comment]
Jimp.decoders appears as undefined
